tests/client/test_client.py

```diff
@@ -165,3 +165,68 @@ def test_get_missing(client, requests_mocker):
 
     # It should explain the problem
     assert "repo1 was not found" in str(error.value)
+
+
+def test_upload_one_file(client, requests_mocker, tmpdir, caplog):
+    """test upload a file to a repo in pulp"""
+    logging.getLogger().setLevel(logging.INFO)
+    caplog.set_level(logging.INFO)
+
+    client._PULP_CHUNK_SIZE = 20
+
+    request_body = {
+        "_href": "/pulp/api/v2/content/uploads/cfb1fed0-752b-439e-aa68-fba68eababa3/",
+        "upload_id": "cfb1fed0-752b-439e-aa68-fba68eababa3",
+    }
+    upload_id = request_body["upload_id"]
+    repo_id = "repo1"
+    import_report = {
+        "result": {},
+        "error": {},
+        "spawned_tasks": [
+            {
+                "_href": "/pulp/api/v2/tasks/7744e2df-39b9-46f0-bb10-feffa2f7014b/",
+                "task_id": "7744e2df-39b9-46f0-bb10-feffa2f7014b",
+            }
+        ],
+    }
+
+    tasks_report = [
+        {"task_id": "7744e2df-39b9-46f0-bb10-feffa2f7014b", "state": "finished"}
+    ]
+
+    somefile = tmpdir.join("some-file.txt")
+    somefile.write("I am the file to be uploaded")
```

Comment:
If you used binary content here instead, the tests would probably catch that issue I mentioned before about not opening the file correctly.

e.g. somefile.write(b'some binary data:\x00\x01\x02')